File: 0997_find_the_town_judge/find_judge.py
from typing import List


def find_judge(N: int, trust: List[List[int]]) -> int:
    """
    Time: O(n)
    Space: O(n)
    """
    if not len(trust):
        return N
    potential = {}
    disqualified = set()
    for a, b in trust:
        if b not in potential:
            potential[b] = 1
        else:
            potential[b] += 1
        disqualified.add(a)
    for k, v in potential.items():
        if k not in disqualified and v == N - 1:
            return k
    return -1

File: 0997_find_the_town_judge/find_judge.go
package main

/**
 * Time: O(n)
 * Space: O(n)
 */
func findJudge(N int, trust [][]int) int {
    votes := make([]int, N + 1)
    for _, item := range(trust) {
        votes[item[0]]--
        votes[item[1]]++
    }
    for i := 1; i < N + 1; i++ {
        if votes[i] == N - 1 {
            return i
        }
    }
    return -1
}

func main() {

}